src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java

```diff
@@ -211,12 +253,28 @@ protected void printSummary() {}
 
   public String produceDts(Compiler compiler) {
     Set<String> provides = new HashSet<>();
-    // TODO: only inspect inputs which were explicitly provided by the user
+    Set<String> transitiveProvides = new HashSet<>();
+    List<String> depgraphRoots = parseDepgraphRoots();
```

Comment:
changed to verify that provided depgraphs have non-empty roots and remove the warning